src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/AvoidUnreliableStreamReadTests.cs

```diff
@@ -0,0 +1,819 @@
+                        await s.ReadExactlyAsync(buffer: buffer, count: count, offset: offset);
+                        await s.ReadExactlyAsync(buffer: buffer, cancellationToken: ct);
+                        await s.ReadExactlyAsync(cancellationToken: ct, buffer: buffer);
+                        await s.ReadExactlyAsync(count: count, cancellationToken: ct, buffer: buffer, offset: offset);
+                    }
+                }
+                """;
+
+            await VerifyCSharpCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task TriviaIsPreserved_OffersFixer_CS()
+        {
+            string source = """
+                using System.IO;
+                using System.Threading;
+
+                class C
+                {
+                    async void M(Stream s, byte[] buffer, int offset, int count, CancellationToken ct)
+                    {
+                        // reticulates the splines
+                        [|s.Read(buffer, 0, buffer.Length)|];
+                        // reticulates the splines
+                        await [|s.ReadAsync(buffer, 0, buffer.Length, ct)|];
+                        // reticulates the splines
+                        [|s.Read(buffer, offset, count)|];
+                        // reticulates the splines
+                        await [|s.ReadAsync(buffer, offset, count, ct)|];
+                    }
+                }
+                """;
+
+            string fixedSource = """
+                using System.IO;
+                using System.Threading;
+
+                class C
+                {
+                    async void M(Stream s, byte[] buffer, int offset, int count, CancellationToken ct)
+                    {
+                        // reticulates the splines
+                        s.ReadExactly(buffer);
+                        // reticulates the splines
+                        await s.ReadExactlyAsync(buffer, ct);
+                        // reticulates the splines
+                        s.ReadExactly(buffer, offset, count);
+                        // reticulates the splines
+                        await s.ReadExactlyAsync(buffer, offset, count, ct);
+                    }
+                }
+                """;
+
+            await VerifyCSharpCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task ReadExactlyNotAvailable_ReportsDiagnostic_CS()
+        {
+            string source = """
+                using System.IO;
+
+                class C
+                {
+                    async void M(Stream s, byte[] buffer)
+                    {
+                        [|s.Read(buffer)|];
+                        await [|s.ReadAsync(buffer)|];
+                    }
+                }
+                """;
+
+            // No code fix because ReadExactly is not available until .NET 7.
+            await VerifyCSharpCodeFixAsync(source, source, ReferenceAssemblies.Net.Net60);
```

Comment:
I was wondering if we want to flag this, then read from the issue `The diagnostic (possibly with a different diagnostic ID) should still fire on older TFMs to report that the call made to Read is unreliable as written.` 
I am ok with having the same diagnostic ID, but I think we should have different message. As if user see a message:
`Avoid unreliable call to 'Stream.Read(byte[])'` as you know, there is no way to avoid this call.

Could have a message something like: `Do not ignore return value of 'Stream.Read(byte[])', less bytes might have read`, what you think? Also tag @gewarren again for better messages.

Reply:
We should provide an example of how to fix the problem without access to `ReadExactly` in the docs. Something like this:
https://github.com/dotnet/runtime/pull/82612/commits/216778b9c5aea6048293d3af05f63af92f9377c8#diff-423ee649d4c4b01f29858b79023a168497214d5c67cc75bc3c5a4b611e12a7beR313-R326

This will help users when they see e.g. `Avoid inexact read with 'System.IO.Ports.SerialStream.Read(byte[], int, int)'` and no code fix is available.

Reply:
> We should provide an example of how to fix the problem without access to ReadExactly in the docs.

Definitely we should. 

> This will help users when they see e.g. `Avoid inexact read with 'System.IO.Ports.SerialStream.Read(byte[], int, int)' and no code fix is available.`

Now with the changed message it looks better to me, thanks!